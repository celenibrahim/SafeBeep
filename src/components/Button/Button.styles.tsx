import { StyleSheet } from "react-native";

export default StyleSheet.create({
    container:{
        backgroundColor:'#0bb1fa',
        borderRadius:10,
        padding:10,
        margin:5,
        alignItems:'center',
        justifyContent:'center',
    },
    text:{
        color:'white',
        fontWeight:'bold',
    }
})