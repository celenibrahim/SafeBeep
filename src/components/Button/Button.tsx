import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import styles from './Button.styles';
const Button = ({text, onPress}: {text: any; onPress: any}) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};
export default Button;
