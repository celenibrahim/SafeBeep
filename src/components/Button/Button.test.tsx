import React from 'react';
import {render} from '@testing-library/react-native';
import Button from './Button';
test('should match with snapshot', () => {
  const comp = render(<Button text={''} onPress={undefined} />);
  expect(comp).toMatchSnapshot();
});
