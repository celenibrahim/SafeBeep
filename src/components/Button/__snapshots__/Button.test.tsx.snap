// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#0bb1fa",
      "borderRadius": 10,
      "justifyContent": "center",
      "margin": 5,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontWeight": "bold",
      }
    }
  />
</View>
`;
